pkg/losetup/losetup.go

```diff
@@ -0,0 +1,102 @@
+package losetup
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"syscall"
+)
+
+const (
+	/*
+	 * IOCTL commands --- we will commandeer 0x4C ('L')
+	 */
+	LOOP_SET_CAPACITY = 0x4C07
+	LOOP_CHANGE_FD    = 0x4C06
+	LOOP_GET_STATUS64 = 0x4C05
+	LOOP_SET_STATUS64 = 0x4C04
+	LOOP_GET_STATUS   = 0x4C03
+	LOOP_SET_STATUS   = 0x4C02
+	LOOP_CLR_FD       = 0x4C01
+	LOOP_SET_FD       = 0x4C00
+	LO_NAME_SIZE      = 64
+	LO_KEY_SIZE       = 32
+	/* /dev/loop-control interface */
+	LOOP_CTL_ADD      = 0x4C80
+	LOOP_CTL_REMOVE   = 0x4C81
+	LOOP_CTL_GET_FREE = 0x4C82
+	SYS_ioctl         = 16
+)
+
+// consider making this a goroutine which pushes the string down the channel.
+func FindLoopDevice() (name string, err error) {
```

Comment:
super go nit: exported functions should have comments that begin with the identifier and are full sentences for godocs.

```go
// FindLoopDevice finds an unused loop device.
//
// TODO: consider...
```

Reply:
What about adding `gofmt` / `golint` to commit hooks (or CI tests) to avoid having to catch these things manually?

Reply:
Yeah, I had opened an issue about that a while ago, but golint currently finds ~300 issues in our code base. Additionally, golint is advisory - some of its advice I'd ignore in certain situations. (I'm wondering if there is a flag to turn off a type of advice...)

Reply:
Apparently not :( https://github.com/golang/lint/issues/263